impeller/geometry/rect_unittests.cc

```diff
@@ -55,5 +55,189 @@ TEST(RectTest, RectMakeSize) {
   }
 }
 
+TEST(RectTest, NormalizePoint) {
+  // Tests finite rects including:
+  // - points at the corners, inside, and outside
+  // - rects that are non-empty or empty through either zero or
+  //   negative width and/or height
+  // - all combinations of integer and scalar rects and points.
+
+  // Tests one rectangle in all 4 combinations of integer and scalar data
+  // and also against NaN point values
+  auto test_one = [](int64_t l, int64_t t, int64_t r, int64_t b,  //
+                     const std::string& rect_desc,                //
+                     int64_t px, int64_t py,                      //
+                     const std::string& pt_desc,                  //
+                     Point expected) {
+    // Scalar point inside Scalar rect
+    ASSERT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(Point(px, py)),
+              expected)
+        << "Point(" << pt_desc << ") in " << rect_desc << " Rect";
+    // Scalar point inside Integer rect
+    ASSERT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(Point(px, py)),
+              expected)
+        << "Point(" << pt_desc << ") in " << rect_desc << " IRect";
+    // Integer point inside Scalar rect
+    ASSERT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(IPoint(px, py)),
+              expected)
+        << "IPoint(" << pt_desc << ") in " << rect_desc << " Rect";
+    // Integer point inside Integer rect
+    ASSERT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(IPoint(px, py)),
+              expected)
+        << "IPoint(" << pt_desc << ") in " << rect_desc << " IRect";
+
+    auto nan = std::numeric_limits<Scalar>::quiet_NaN();
+    auto nan_x = Point(nan, py);
+    auto nan_y = Point(px, nan);
+    auto nan_p = Point(nan, nan);
+    // Nan Scalar point inside Scalar and integer rects
+    ASSERT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_x), Point())
+        << "Point(NaN x) in " << rect_desc << " Rect";
+    ASSERT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_y), Point())
+        << "Point(NaN y) in " << rect_desc << " Rect";
+    ASSERT_EQ(Rect::MakeLTRB(l, t, r, b).NormalizePoint(nan_p), Point())
+        << "Point(NaN x&y) in " << rect_desc << " Rect";
+    ASSERT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_x), Point())
+        << "Point(NaN x) in " << rect_desc << " Rect";
+    ASSERT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_y), Point())
+        << "Point(NaN y) in " << rect_desc << " Rect";
+    ASSERT_EQ(IRect::MakeLTRB(l, t, r, b).NormalizePoint(nan_p), Point())
+        << "Point(NaN x&y) in " << rect_desc << " Rect";
```

Comment:
You should change this to EXPECT_EQ so the test always runs to completion.  Probably all of them I'd have to double check.

Reply:
I find a lot of inconsistency on that in the various tests. I can see your point about running to completion, but in the end all of the errors have to be fixed anyway and they are often due to the same mistake. So, is it spam or is it a full roster of errors?

I suppose given that none of these ASSERT statements can actually terminate the test (since they are inside local closures), EXPECT would be consistent with the fact that the test will complete its various calls to test() regardless of whether an individual call to test() terminates early.

Reply:
They are strict about this internally and I ran into it the other day which reminded me not to sleep on it.

Reply:
In doing a global replace I discovered that the `NEAR` macros don't have `EXPECT` equivalents. I'll file an issue so that we can apply the same logic to those kinds of tests...

(Filed: https://github.com/flutter/flutter/issues/137873)

Reply:
googltest has `EXPECT_NEAR`

Reply:
For single values, but the Impeller macros are for whole objects.